src/Uno.UI/UI/Xaml/Media/Animation/Animators/GPUFloatValueAnimator.iOSmacOS.cs

```diff
@@ -37,10 +38,10 @@ internal class GPUFloatValueAnimator : IValueAnimator
 		private float _to;
 		private float _from;
 		private long _duration;
-		private IEnumerable<IBindingItem> _bindingPath;
-		private FloatValueAnimator _valueAnimator;
-		private UnoCoreAnimation _coreAnimation;
-		private IEasingFunction _easingFunction;
+		private IEnumerable<IBindingItem>? _bindingPath;
+		private FloatValueAnimator? _valueAnimator;
```

Comment:
This one is only null after dispose. It's probably an issue if the object is dealt with after it was disposed.

Usually, these cases are annotated as non-null, then in `Dispose`, it's done with null suppression, `_valueAnimator = null!`.

Reply:
Alright, I will follow this. 

"This one is only null after dispose. It's probably an issue if the object is dealt with after it was disposed."
I believe that's what's happening, that for some reason, during the navigation, we are dealing with a disposed Animation object and reaching a point where some fields are already `null`

Reply:
I think that's the real bug then, the object is used in a disposed state. But as mentioned in https://github.com/unoplatform/uno/pull/17558#discussion_r1679814958, we could tolerate for use-after-dispose if it ends up complicated to fix it the right way

Reply:
@Youssef1313, thanks for your input. Yes, this is a workaround since we have not been able to reproduce the actual error but found that by doing these changes we could prevent crashes or make it crash in a more evident place

Reply:
An additional information was provided for the nullref, it's failing at : 

https://github.com/unoplatform/uno/blob/161c3c1959ecaa00dd55c658d099b4c49f6feeed/src/Uno.UI/UI/Xaml/Media/Animation/Animators/GPUFloatValueAnimator.iOSmacOS.cs#L436-L438